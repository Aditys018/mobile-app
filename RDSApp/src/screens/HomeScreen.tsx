import React from 'react';
import {View, Text} from 'react-native';
import {ScreenViewContainer} from '../styles/GlobalStyle';

const HomeScreen = () => {
  return (
    <View style={ScreenViewContainer.container}>
      <Text style={ScreenViewContainer.text_view}>Home Screen </Text>
    </View>
  );
};

export default HomeScreen;
