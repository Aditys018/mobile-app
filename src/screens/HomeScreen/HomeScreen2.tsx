import { Button, StyleSheet, Text, View } from 'react-native';
import React, { useState } from 'react';
import Strings from '../../i18n/en';
import OOOForm from '../../components/OOO/OOOForm';

const HomeScreen2 = (): JSX.Element => {
  const [isFormVisible, setIsFormVisible] = useState<boolean>(false);
  const [fromDate, setFromDate] = useState<number>(Date.now());
  const [toDate, setToDate] = useState<number>(Date.now());
  const [description, setDescription] = useState<string>('');

  const handleButtonPress = () => {
    setIsFormVisible((prev) => !prev);
  };

  const handleFormSubmit = () => {
    // patch api call (it should give loading , failed, error and success messages)
    console.log(fromDate, toDate, description); // was giving lint issues, after api call please remove this log
    setIsFormVisible((prev) => !prev);
  };

  return (
    <View>
      <Text>You are Idle</Text>
      <Button
        title={Strings.UPDATE_STATUS_TO_OOO}
        onPress={handleButtonPress}
      />
      {isFormVisible && (
        <OOOForm
          fromDate={fromDate}
          toDate={toDate}
          description={description}
          setToDate={setToDate}
          setFromDate={setFromDate}
          setDescription={setDescription}
          handleFormSubmit={handleFormSubmit}
        />
      )}
    </View>
  );
};

export default HomeScreen2;
