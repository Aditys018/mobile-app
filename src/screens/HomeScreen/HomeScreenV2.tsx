import { Text, View } from 'react-native';
import React, { useState, useEffect } from 'react';
import Strings from '../../i18n/en';
import OOOForm from '../../components/OOO/OOOForm';
import { AuthScreenButton } from '../AuthScreen/Button';
import { getUsersStatus, submitOOOForm } from '../AuthScreen/Util';

const HomeScreenV2 = (): JSX.Element => {
  const [isFormVisible, setIsFormVisible] = useState<boolean>(false);
  const [fromDate, setFromDate] = useState<number>(Date.now());
  const [toDate, setToDate] = useState<number>(Date.now());
  const [description, setDescription] = useState<string>('');
  const [status, setStatus] = useState('');

  useEffect(() => {
    fetchData();
  }, []);
  const fetchData = async () => {
    const userStatus = await getUsersStatus();
    setStatus(userStatus);
  };
  const handleButtonPress = () => {
    setIsFormVisible((prev) => !prev);
  };

  const handleFormSubmit = async () => {
    // patch api call (it should give loading , failed, error and success messages)
    let data = {
      currentStatus: {
        from: fromDate,
        message: description,
        state: 'OOO',
        until: toDate,
        updateAt: 987937932,
      },
    };
    const updateStatus = await submitOOOForm(data);
    console.log({ data23131: updateStatus }); // was giving lint issues, after api call please remove this log
    setIsFormVisible((prev) => !prev);
  };

  return (
    <View>
      <Text>{status}</Text>
      <View style={{ marginTop: 20 }}>
        <AuthScreenButton
          text={Strings.UPDATE_STATUS_TO_OOO}
          onPress={handleButtonPress}
        />
      </View>

      {isFormVisible && (
        <OOOForm
          fromDate={fromDate}
          toDate={toDate}
          description={description}
          setToDate={setToDate}
          setFromDate={setFromDate}
          setDescription={setDescription}
          handleFormSubmit={handleFormSubmit}
        />
      )}
    </View>
  );
};

export default HomeScreenV2;
