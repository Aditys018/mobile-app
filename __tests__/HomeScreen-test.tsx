/**
 * @format
 */

import 'react-native';
import React from 'react';
import HomeScreen from '../src/screens/HomeScreen/HomeScreen';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

import {render, fireEvent} from '@testing-library/react-native';

import Strings from '../src/i18n/en';

it('renders correctly', () => {
  renderer.create(<HomeScreen />);
});

it('snapshot should match', () => {
  const tree = renderer.create(<HomeScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('If a user clicks on Mark Yourself As OOO', () => {
  const {getByText, getAllByText} = render(<HomeScreen />);

  fireEvent.press(getByText(Strings.OOOBtn1_Text));

  const renderedElements = getAllByText(Strings.OOOStatus_Text);

  expect(renderedElements).toHaveLength(1);
});

test('If a user clicks on Mark Yourself Active Again', () => {
  const {getByText, getAllByText} = render(<HomeScreen />);

  fireEvent.press(getByText(Strings.OOOBtn1_Text));

  fireEvent.press(getByText(Strings.OOOBtn2_Text));

  const renderedElements = getAllByText(Strings.Active_Text);

  expect(renderedElements).toHaveLength(1);
});

test('If a user clicks on change status to Idle', () => {
  const {getByText, getAllByText} = render(<HomeScreen />);

  fireEvent.press(getByText(Strings.OOOBtn1_Text));

  const renderedElements = getAllByText(Strings.OOOStatus_Text);

  expect(renderedElements).toHaveLength(1);
});

test('If a user clicks on change status to Active', () => {
  const {getByText, getAllByText} = render(<HomeScreen />);

  fireEvent.press(getByText(Strings.OOOBtn1_Text));

  const renderedElements = getAllByText(Strings.OOOStatus_Text);

  expect(renderedElements).toHaveLength(1);
});
