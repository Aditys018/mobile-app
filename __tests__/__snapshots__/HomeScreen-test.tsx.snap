// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot should match 1`] = `
Array [
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "backgroundColor": "#1D1283",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "backgroundColor": "#1D1283",
          "justifyContent": "flex-start",
          "padding": 15,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../assets/rdsLogo.png",
          }
        }
        style={
          Object {
            "height": 45,
            "width": 45,
          }
        }
      />
    </View>
  </RCTSafeAreaView>,
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "padding": 8,
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "color": "#0034a5",
            "fontSize": 35,
            "fontWeight": "bold",
            "marginBottom": 30,
            "textAlign": "center",
          }
        }
      >
        I am doing a task
      </Text>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "borderColor": "#e49504",
            "borderRadius": 20,
            "borderWidth": 2,
            "marginBottom": 30,
            "marginLeft": "auto",
            "marginRight": "auto",
            "marginTop": 20,
            "opacity": 1,
            "paddingBottom": 20,
            "paddingLeft": 40,
            "paddingRight": 40,
            "paddingTop": 20,
            "textAlign": "center",
            "userSelect": "none",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#e49504",
              "fontSize": 20,
            }
          }
        >
          change status to idle
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#0034a5",
            "fontSize": 20,
            "fontWeight": "500",
            "textAlign": "center",
          }
        }
      >
        Mark Yourself as OOO
      </Text>
    </View>
  </View>,
]
`;
